### DISCUSSION TITLE
Store more theme data in themes/_list.json

### DISCUSSION THREAD

**Discussion by lukew3** [monkeytypegame/monkeytype#4345] [Ideas]
The 4 main colors and theme name are already stored in the [`/frontend/static/themes/_list.json`](https://github.com/monkeytypegame/monkeytype/blob/master/frontend/static/themes/_list.json) file but in order to load each theme, a separate css file needs to be loaded. This means that in offline mode themes cannot be changed, even though most of the data is already cached.

I propose that all color data is added to the `_list.json` and color variables are updated with javascript instead of fetching a css file if not necessary. If a theme requires extended css beyond just changing root color variables, an extended css property flag can be set in the `_list.json`, meaning that a css file will have to be retrieved. Adding all colors to the list will result in the list file becoming about twice it's size, from ~24kb to ~50kb. Since every theme requires the same 10 color variables, naming colors in the list is not necessary and each theme can be represented as an array of color codes.
If a user is offline and loads a partial theme, a grey notification should show that says, "Extended theme styles could not be loaded offline"

I also think that the themes list should be a hashset instead of a list and be organized alphabetically for readers. The key would be the theme name and the value would be a list of color values.

**Comment by ridemountainpig** [comment]
Hi @lukew3 , I have collect all theme together and create a api to get all monkeytype theme, you can just get all theme data by this url, you can try it.

The api return all theme data like this
<img width="1297" alt="CleanShot 2023-07-18 at 20 50 09@2x" src="https://github.com/monkeytypegame/monkeytype/assets/92412722/bbad1dd4-362c-473e-bc0c-504fd0528cfc">

https://monkeytype-readme.repl.co/mr-command/theme
